src/dc_job.rs

```diff
@@ -792,23 +789,23 @@ unsafe fn dc_add_smtp_job(
             {
                 (*(*mimefactory).msg).id
             } else {
-                0i32 as libc::c_uint
+                0
             }) as libc::c_int,
-            (*param).packed,
-            0i32,
+            Some(param),
+            0,
         );
         success = 1i32
     }
-    dc_param_unref(param);
     free(recipients as *mut libc::c_void);
     free(pathNfilename as *mut libc::c_void);
     return success;
 }
+
 pub unsafe fn dc_job_add(
     context: &Context,
     action: libc::c_int,
     foreign_id: libc::c_int,
-    param: *const libc::c_char,
+    param: Option<Params>,
```

Comment:
Here again the code is simpler if it just accepts plain params without the option IMHO.  `Params::new()` isn't much harder to write than `None` (though I guess it is less efficient)

Reply:
fixed